livekit-plugins/livekit-plugins-deepgram/livekit/plugins/deepgram/stt.py

```diff
@@ -318,132 +451,159 @@ async def keepalive_task(ws: aiohttp.ClientWebSocketResponse):
+                            AudioEnergyFilter.State.SPEAKING,
+                        ):
+                            if last_frame:
+                                frames.extend(
+                                    audio_bstream.write(last_frame.data.tobytes())
+                                )
+                                last_frame = None
+                            frames.extend(audio_bstream.write(data.data.tobytes()))
+                        elif state == AudioEnergyFilter.State.END:
+                            # no need to buffer as we have cooldown period
+                            frames = audio_bstream.flush()
+                            has_ended = True
+                        elif state == AudioEnergyFilter.State.SILENCE:
+                            # buffer the last silence frame, since it could contain beginning of speech
+                            # TODO: improve accuracy by using a ring buffer with longer window
+                            last_frame = data
+                    elif isinstance(data, self._FlushSentinel):
                         frames = audio_bstream.flush()
                         has_ended = True
-                    elif state == AudioEnergyFilter.State.SILENCE:
-                        # buffer the last silence frame, since it could contain beginning of speech
-                        # TODO: improve accuracy by using a ring buffer with longer window
-                        last_frame = data
-                elif isinstance(data, self._FlushSentinel):
-                    frames = audio_bstream.flush()
-                    has_ended = True
-
-                for frame in frames:
-                    self._audio_duration_collector.push(frame.duration)
-                    await ws.send_bytes(frame.data.tobytes())
-
-                    if has_ended:
-                        self._audio_duration_collector.flush()
-                        await ws.send_str(SpeechStream._FINALIZE_MSG)
-                        has_ended = False
-
-            # tell deepgram we are done sending audio/inputs
-            closing_ws = True
-            await ws.send_str(SpeechStream._CLOSE_MSG)
+
+                    for frame in frames:
+                        self._audio_duration_collector.push(frame.duration)
+                        await ws.send_bytes(frame.data.tobytes())
+
+                        if has_ended:
+                            self._audio_duration_collector.flush()
+                            await ws.send_str(SpeechStream._FINALIZE_MSG)
+                            has_ended = False
+
+                # tell deepgram we are done sending audio/inputs
+                closing_ws = True
+                await ws.send_str(SpeechStream._CLOSE_MSG)
+            except asyncio.CancelledError:
+                pass
+            except Exception as e:
+                logger.error("unexpected error in send_task %s", e)
+                return
```

Comment:
We should not catch errors, errors must be propagated to the task running the `_run` method (Since the retry logic is handled on the base class)